### ISSUE TITLE
`PrimitiveArrayDeserializers$ByteDeser.deserialize` ignores `DeserializationProblemHandler` for invalid Base64 content

### ISSUE THREAD

**Issue by or-else** [FasterXML/jackson-databind#3784]
**Describe the bug**

`PrimitiveArrayDeserializers$ByteDeser.deserialize` throws `InvalidFormatException` instead of calling one of the `DeserializationProblemHandler` methods, probably `handleWeirdStringValue`.

```
InvalidFormatException: Cannot access contents of TextNode as binary due to broken 
Base64 encoding: Illegal character '␡'
```

**Version information**

Tested in 2.13.5, but the bug is definitely present in 2.15

**To Reproduce**

```java
class Test {
   public byte[] data;
}
```
json:
```json
{"data": "not a valid base64 string"}
```

**Expected behavior**

I expect deserializer to call `handleWeirdStringValue` (or another method of `DeserializationProblemHandler`, if that's more appropriate) instead of just throwing `InvalidFormatException`.

**Comment by or-else** [comment]
The actual bug is in this line:

https://github.com/FasterXML/jackson-databind/blob/2.15/src/main/java/com/fasterxml/jackson/databind/deser/std/PrimitiveArrayDeserializers.java#L470

It expects `StreamReadException ` while `InvalidFormatException` is thrown.

**Comment by cowtowncoder** [comment]
Yes, that makes sense. Will need to write a unit test to properly reproduce the problem, ensure handler is called etc.

**Comment by or-else** [comment]
Would it be possible to backport it to 2.13? I'm stuck with 2.13 because I need Android API 24. 2.14 requires API 26.

**Comment by cowtowncoder** [comment]
@or-else Hmmh. Probably, although I am not sure there will be new releases of 2.13. But fix itself could go there.

**Comment by or-else** [comment]
@cowtowncoder Would be nice to have a 2.13.6

**Comment by cowtowncoder** [comment]
Alas, it takes multiple hours to publish a full patch release so there typically has to be a few more important fixes.
So it all comes down to availability of time & where it is spent.

**Comment by cowtowncoder** [comment]
@or-else Actually, I realized that I am not sure how to reproduce this -- exception message suggests you are not actually reading it directly from JSON (which may actually work as expected) but probably first read content as `JsonNode` and then try binding? I can try to recreate it but if I can't I may need help here.

**Comment by cowtowncoder** [comment]
Ok yes; regular read actually worked fine. It's only if reading/converting from `JsonNode` where different exception is thrown. I was able to fix this for 2.13 branch just in case there might be a new release; but will regardless be in next 2.14 and 2.15 releases.

**Comment by or-else** [comment]
> first read content as JsonNode and then try binding? 

Correct.
